zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -336,88 +344,79 @@ mod test {
-            gas: gas.map(Word::from).unwrap_or(minimal_gas),
-            gas_price: gas_price.or(Some(two_gwei)),
+            value,
+            gas,
+            gas_price: Some(gas_price),
             input: calldata.into(),
             ..Default::default()
         }
     }
 
     #[test]
     fn begin_tx_gadget_simple() {
-        // Transfer 1 ether, successfully
-        test_ok(mock_tx(None, None, None, vec![]), true);
+        // Transfer 0.1 ether, successfully
+        test_ok(
+            mock_tx(*POINT_ONE_ETHER, *MINIMAL_GAS, *TWO_GWEI, vec![]),
+            true,
+        );
 
-        // Transfer 1 ether, tx reverts
-        test_ok(mock_tx(None, None, None, vec![]), false);
+        // Transfer 0.1 ether, tx reverts
+        test_ok(
+            mock_tx(*POINT_ONE_ETHER, *MINIMAL_GAS, *TWO_GWEI, vec![]),
+            false,
+        );
 
         // Transfer nothing with some calldata
         test_ok(
-            mock_tx(None, Some(21086), None, vec![1, 2, 3, 4, 0, 0, 0, 0]),
+            mock_tx(
+                Word::zero(),
+                Word::from(21086),
+                *TWO_GWEI,
+                vec![1, 2, 3, 4, 0, 0, 0, 0],
+            ),
             false,
         );
     }
 
     #[test]
     fn begin_tx_gadget_rand() {
-        let point_one_ether = Word::from(10u8).pow(Word::from(17u8));
+        let random_amount = Word::from_little_endian(&rand_bytes(32)) % *POINT_ONE_ETHER;
+        let random_gas_price = Word::from(rand_range(0..47619047619047u64));
+
+        // If this test fails, we want these values to appear in the CI logs.
+        dbg!(random_amount, random_gas_price);
 
         // Transfer random ether, successfully
         test_ok(
-            mock_tx(
-                Some(Word::from_little_endian(&rand_bytes(32)) % point_one_ether),
-                None,
-                None,
-                vec![],
-            ),
+            mock_tx(random_amount, *MINIMAL_GAS, *TWO_GWEI, vec![]),
             true,
         );
 
         // Transfer nothing with random gas_price, successfully
         test_ok(
-            mock_tx(
-                None,
-                None,
-                Some(Word::from(rand_range(0..47619047619047u64))),
-                vec![],
-            ),
+            mock_tx(Word::zero(), *MINIMAL_GAS, random_gas_price, vec![]),
             true,
         );
 
         // Transfer random ether, tx reverts
         test_ok(
-            mock_tx(
-                Some(Word::from_little_endian(&rand_bytes(32)) % point_one_ether),
-                None,
-                None,
-                vec![],
-            ),
+            mock_tx(random_amount, *MINIMAL_GAS, *TWO_GWEI, vec![]),
             false,
         );
 
         // Transfer nothing with random gas_price, tx reverts
         test_ok(
-            mock_tx(
-                None,
-                None,
-                Some(Word::from(rand_range(0..476190476190476u64))),
```

Comment:
`476190476190476` comes from <code>10<sup>19</sup> // 21000</code>, but it should be <code>10<sup>19</sup> // 21006</code> (`476054460630296`), so this could be the reason the test failed.

Reply:
Let's try to also annotate the origin of the value in a comment or the expr directly.

Reply:
I just added a function that computes the gas price.